packages/react/src/components/AccountSettings/ConfigureTOTP/ConfigureTOTP.tsx

```diff
@@ -0,0 +1,167 @@
+        const username = totpUsername || user?.username;
+        const totpCode = getTotpCodeURL(totpIssuer, username, newSecretKey);
+        const qrCodeImageSource = await QRCode.toDataURL(totpCode);
+
+        if (!hasInit.current) {
+          setSecretKey(newSecretKey);
+          setQrCode(qrCodeImageSource);
+          hasInit.current = false;
+        }
+      } catch (e) {
+        logger.error(e);
+      }
+    },
+    [totpIssuer, totpUsername]
+  );
+
+  React.useEffect(() => {
+    if (user && !hasInit.current) {
+      generateQRCode(user);
+    }
+  }, [generateQRCode, user]);
+
+  // translations
+  const confirmText = translate('Confirm');
+  const copyCodeText = translate('Copy Secret Code');
+
+  // event handlers
+  const handleChange = React.useCallback(
+    (event: React.ChangeEvent<HTMLInputElement>) => {
+      event.preventDefault();
+      const { name, value } = event.target;
+      setFormValues((formValues) => ({ ...formValues, [name]: value }));
+    },
+    []
+  );
+
+  const runVerifyTotpToken = React.useCallback(
+    async ({ user, code }: { user: AmplifyUser; code: string }) => {
+      if (errorMessage) {
+        setErrorMessage(null);
+      }
+      try {
+        await verifyTOTPToken({ user, code });
+
+        onSuccess?.(); // notify success to the parent
+      } catch (e) {
+        const error = e as Error;
+        if (error.message) {
+          setErrorMessage(error.message);
+        }
+
+        onError?.(error); // notify error to the parent
+      }
+    },
+    [errorMessage, onError, onSuccess]
+  );
+
+  const handleSubmit = React.useCallback(
+    (event: React.FormEvent<HTMLFormElement>) => {
+      event.preventDefault();
+      const { code } = formValues;
+      runVerifyTotpToken({ user, code });
+    },
+    [user, formValues, runVerifyTotpToken]
+  );
+
+  const handleCopy = React.useCallback(() => {
+    navigator.clipboard.writeText(secretKey);
+  }, [secretKey]);
+
+  /** Return null if user isn't authenticated in the first place */
+  if (!user) {
+    logger.warn('<ConfigureTOTP /> requires user to be authenticated.');
+    return null;
+  }
+
+  /** Return null if Auth.getCurrentAuthenticatedUser is still in progress  */
+  if (isLoading) {
+    return null;
+  }
+
+  return (
+    <View as="form" onSubmit={handleSubmit}>
+      <Flex direction="column" alignItems="center">
+        {qrCode ? (
+          <SecretKeyQRCode
+            data-amplify-qrcode
+            src={qrCode}
+            alt="qr code"
+            width="228px"
+            height="228px"
+          />
+        ) : null}
+        <CopySecretKey alignSelf="stretch" onClick={handleCopy}>
+          {copyCodeText}
+        </CopySecretKey>
+        <ConfirmationCode
+          alignSelf="stretch"
+          isRequired
+          label="Confirmation Code"
+          name="code"
+          onChange={handleChange}
+          placeholder="Code"
+          value={formValues.code}
+          width="100%"
```

Comment:
Just curious, do we need both `alignSelf="stretch"` and `width="100%"`?

Reply:
We do not, good catch.
```suggestion
```